WooCommerce/Classes/View Modifiers/View+MediaSourceActionSheet.swift

```diff
@@ -3,50 +3,71 @@ import SwiftUI
 
     init(showsActionSheet: Binding<Bool>,
+         sourceOptions: [MediaPickingSource],
          selectMedia: @escaping (MediaPickingSource) -> Void) {
         self.showsActionSheet = showsActionSheet
+        self.sourceOptions = sourceOptions
         self.selectMedia = selectMedia
     }
 
     func body(content: Content) -> some View {
         content
             .confirmationDialog(Text(Localization.title), isPresented: showsActionSheet, actions: {
-                if UIImagePickerController.isSourceTypeAvailable(.camera) {
-                    Button(Localization.camera) {
-                        selectMedia(.camera)
+                ForEach(sourceOptions) { source in
+                    switch source {
+                    case .camera:
+                        if UIImagePickerController.isSourceTypeAvailable(.camera) {
+                            Button(Localization.camera) {
+                                selectMedia(.camera)
+                            }
+                        }
+                    case .photoLibrary:
+                        Button(Localization.photoLibrary) {
+                            selectMedia(.photoLibrary)
+                        }
+                    case .siteMediaLibrary:
+                        Button(Localization.siteMediaLibrary) {
+                            selectMedia(.siteMediaLibrary)
+                        }
+                    case .productMedia(let productID):
+                        Button(Localization.productMedia) {
+                            selectMedia(.productMedia(productID: productID))
+                        }
                     }
                 }
-
-                Button(Localization.photoLibrary) {
-                    selectMedia(.photoLibrary)
-                }
-
-                Button(Localization.siteMediaLibrary) {
-                    selectMedia(.siteMediaLibrary)
-                }
             })
     }
 }
 
 private extension MediaSourceActionSheet {
     enum Localization {
         static let title = NSLocalizedString(
-            "Select Media Source",
+            "mediaSourceActionSheet.title",
+            value: "Select Media Source",
             comment: "Title of the media picker action sheet to select a source."
         )
         static let camera = NSLocalizedString(
-            "Take a photo",
+            "mediaSourceActionSheet.camera",
+            value: "Take a photo",
             comment: "Menu option for taking an image or video with the device's camera."
         )
         static let photoLibrary = NSLocalizedString(
-            "Choose from device",
+            "mediaSourceActionSheet.photoLibrary",
+            value: "Choose from device",
             comment: "Menu option for selecting media from the device's photo library."
         )
+        static let productMedia = NSLocalizedString(
+            "mediaSourceActionSheet.productMedia",
+            value: "Choose from product media",
```

Comment:
Nit: Would it be clearer to use "Choose an existing product photo"?

Reply:
Good suggestion. I changed it [here](https://github.com/woocommerce/woocommerce-ios/pull/13540/commits/ce00414178858d1c6f9823c16b10eb02c8a3b503) in this PR. https://github.com/woocommerce/woocommerce-ios/pull/13540